spec/integration/provider/yumrepo_spec.rb

```diff
@@ -0,0 +1,124 @@
+      apply_with_error_check(manifest)
+      expect_file_mode(File.join(@yumrepo_dir, super_creative + '.repo'), "644")
+      expect(File.read(File.join(@yumrepo_dir, super_creative + '.repo'))).
+        to match(expected_match + "\n")
+    end
+
+    it 'should remove a managed yumrepo entry' do
+      apply_with_error_check(manifest)
+      manifest = "#{type_under_test} { '#{super_creative}':
+                    ensure => absent,
+                    target  => '#{@yumrepo_file}' }"
+      apply_with_error_check(manifest)
+      expect(File.read(File.join(@yumrepo_dir, super_creative + '.repo'))).
+        to be_empty
+    end
+
+    it 'should update a managed yumrepo entry' do
+      apply_with_error_check(manifest)
+      manifest = "#{type_under_test} { '#{super_creative}':
+                    baseurl => 'http://#{super_creative}.updated',
+                    target  => '#{@yumrepo_file}' }"
+      apply_with_error_check(manifest)
+      expect(File.read(File.join(@yumrepo_dir, super_creative + '.repo'))).
+        to match(expected_match + ".updated\n")
+    end
+
+    it 'should create all properties of a yumrepo entry' do
+      manifest = "#{type_under_test} { '#{super_creative}':
+                    baseurl => 'http://#{super_creative}',
+                    target  => '#{@yumrepo_file}' }"
+      apply_with_error_check(manifest)
+      expect(File.read(File.join(@yumrepo_dir, super_creative + '.repo'))).
+        to match("\[#{super_creative}\]")
+    end
+
+    # The unit-tests cover all properties
+    #   and we have to hard-code the "should" values here.
+    # Puppet::Type.type(:yumrepo).validproperties contains the full list
+    #   but we can't get the property "should" values from the yumrepo-type
+    #   without having an instance of type, which is what yumrepo defines...
+    #   Just cover the most probable used properties.
+    properties = {"bandwidth"      => "42M",
+                  "baseurl"        => "http://er0ck",
+                  "cost"           => "42",
+                  "enabled"        => "YeS",
+                  "exclude"        => "er0ckSet2.0",
+                  "failovermethod" => "roundrobin",
+                  "include"        => "https://er0ck",
+                  "mirrorlist"     => "https://er0ckMirr0r.co",
+                  "priority"       => 99,
+                  "retries"        => 413189,
+                  "timeout"        => 666
+    }
+
+    properties.each do |property_key, property_value|
+      it "should create an entry with #{property_key} property" do
+        manifest = "#{type_under_test} { '#{super_creative}':
+                      #{property_key} => '#{property_value}',
+                              target  => '#{@yumrepo_file}' }"
+        apply_with_error_check(manifest)
+        expect(File.read(File.join(@yumrepo_dir, super_creative + '.repo'))).
+          to match(/^#{property_key}=#{property_value}$/)
```

Comment:
do we need to `Regexp.escape(property_value)`, e.g. when the value is `https://er0ck`?